shuup/core/api/basket.py

```diff
@@ -543,6 +546,26 @@ def _handle_setting_address(self, request, attr_field):
             data = BasketSerializer(request.basket, context=self.get_serializer_context()).data
             return Response(data, status=status.HTTP_200_OK)
 
+    @schema_serializer_class(ExtraDataSerializer)
+    @detail_route(methods=['post'])
+    def set_extra_data(self, request, *args, **kwargs):
+        self.process_request()
+        serializer = ExtraDataSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        if serializer.is_valid():
+            if not serializer.validated_data:
+                return Response({"error": "invalid key"}, status=status.HTTP_400_BAD_REQUEST)
+
+            extra_data = {}
+            for k, v in request.data.items():
```

Comment:
you should probably iterate over `serializer.validated_data` since it will parse the values correctly accordingly to the serializer field types